Model/Kount/EnsConfig.php

```diff
@@ -0,0 +1,361 @@
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     * @param ScopeConfigInterface $scopeConfig
+     * @param OrderInterface $order
+     * @param OrderRepositoryInterface $orderRepository
+     * @param BraintreeAdapter $braintreeAdapter
+     * @param TransactionFactory $transactionFactory
+     * @param CreditmemoFactory $creditmemoFactory
+     * @param CreditmemoService $creditmemoService
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        ScopeConfigInterface $scopeConfig,
+        OrderInterface $order,
+        OrderRepositoryInterface $orderRepository,
+        BraintreeAdapter $braintreeAdapter,
+        TransactionFactory $transactionFactory,
+        CreditmemoFactory $creditmemoFactory,
+        CreditmemoService $creditmemoService
+    ) {
+        $this->scopeConfig = $scopeConfig;
+        $this->storeManager = $storeManager;
+        $this->order = $order;
+        $this->braintreeAdapter = $braintreeAdapter;
+        $this->transactionFactory = $transactionFactory;
+        $this->creditmemoFactory = $creditmemoFactory;
+        $this->creditmemoService = $creditmemoService;
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * @return array
+     */
+    public function getAllowedIps(): array
+    {
+        $ips = $this->scopeConfig->getValue(self::CONFIG_ALLOWED_IPS);
+        return $ips ? explode(',', $ips) : [];
+    }
+
+    /**
+     * This method will check if a given IP (as a string) falls into a valid range, e.g "192.168.0.0/255".
+     *
+     * @param $ip
+     * @param $range
+     * @return bool
+     */
+    public function isIpInRange($ip, $range): bool
+    {
+        // If no "range" is set, add the full range of 255.
+        if (strpos($range, '/') === false) {
+            $range .= '/255';
+        }
+
+        // $range is in IP/CIDR format eg 127.0.0.1/255
+        list($range, $netmask) = explode('/', $range, 2); // Get the starting IP and the netmask of the IP range.
+
+        $range_decimal = ip2long($range);
+        $ip_decimal = ip2long($ip);
+        $wildcard_decimal = (2 ** (32 - $netmask)) - 1;
+        $netmask_decimal = ~ $wildcard_decimal;
+
+        return (($ip_decimal & $netmask_decimal) === ($range_decimal & $netmask_decimal));
+    }
+
+    /**
+     * @param string $remoteAddress
+     * @return bool
+     */
+    public function isAllowed(string $remoteAddress): bool
+    {
+        $allowedIps = $this->getAllowedIps();
+
+        if (!$allowedIps) {
+            return true;
+        }
+
+        foreach ($allowedIps as $allowedIp) {
+            if ($this->isIpInRange($remoteAddress, $allowedIp)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $merchantId
+     * @return bool
+     */
+    public function validateMerchantId($merchantId): bool
```

Comment:
Typed to `int`? (based on line 165)

Reply:
I'm not entirely sure on the correct format, as Kount don't seem to give that away. I only know its a 6 digit "number" e.g. `<events merchant="602920" total="1">` so most probably best to type hint as a string? :)